commands/commands.go

```diff
@@ -141,15 +132,49 @@ func RunWithTimeout(c *Command, fields log.Fields) error {
 	log.Debugf("%s.RunWithTimeout start", c.Name)
 	c.setUpCmd(fields)
 	defer c.closeLogs()
-	log.Debugf("%s.Cmd.Start", c.Name)
+	log.Debugf("%s.Cmd start", c.Name)
+	ctx, cancel := context.WithTimeout(context.Background(), c.TimeoutDuration)
+	defer cancel()
 	if err := c.Cmd.Start(); err != nil {
-		log.Errorf("Unable to start %s: %v", c.Name, err)
+		log.Errorf("unable to start %s: %v", c.Name, err)
+		return err
+	}
+	go func() {
+		select {
+		case <-ctx.Done():
+			// if the context was canceled we don't want to kill the
+			// process because it's already gone
+			if ctx.Err().Error() == "context deadline exceeded" {
```

Comment:
Hopefully Go is good about not changing error text without considering it a breaking change. I have no idea whether it is or not, but I've been burned in other languages.